### PR TITLE
Add lookup for network provider by MCC and MNC

### PR DESCRIPTION
**Pull request by ConnorRigby**

### CODE REVIEW

File: lib/vintage_net_qmi/mcc_mnc.ex [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        iex> VintageNetQMI.MCCMNC.provider("310", "410")
+        "AT&T"
+
+        iex> VintageNetQMI.MCCMNC.provider("000", "000")
+        nil
+  """
+  @spec provider(String.t(), String.t()) :: String.t() | nil
+  for {mcc, mnc, provider} <- @providers do
+    def provider(unquote(mcc), unquote(mnc)), do: unquote(provider)
```
Comment (fhunleth) [inline lib/vintage_net_qmi/mcc_mnc.ex:29]: Can this be done in a way that makes the resulting `.beam` smaller?

While it's not large, it feels a little silly to have such a lightly used database in DRAM all the time. Here are some ideas:
1. Figure out a way to make it small enough to not care. I did a quick test for making making the key be an integer representation of the mcc/mnc. That reduced the compressed .beam file from 79452 to 53472. Maybe there's a way to do even better. I didn't look if the string values were highly repetitive and Erlang wasn't deduplicating them.
2. Keep the .csv file or a processed version of it in the `priv` directory. Look the provider up when needed.

Comment (ConnorRigby) [reply lib/vintage_net_qmi/mcc_mnc.ex]: updated it to just do the lookup at runtime. keepin it simple

**Comment by fhunleth** [comment]
@ConnorRigby I force pushed to get rid of some extraneous dependency bumps.

I also added a commit to avoid loading the whole csv file into memory and performing computations on it. I like the route of keeping it on disk, but it seemed not ideal to load it all and do all of the in memory computations at runtime when only one line was needed.

Since we're storing this on embedded devices, I wonder if it's worth a little bit more preprocessing of the csv file to make it smaller. Or perhaps it's worth a compile-time sanity check to avoid surprise run-time exceptions.

**Comment by fhunleth** [comment]
Looking more at the csv file. We could chop out MCC and MNC and just use the PLMN. Region and country seem redundant with ISO. Dupes could be removed. There's also some text encoding issue - look for Finland entries. Keeping as a text file seems fine with me since I'm not sure how a binary file would be any smaller. We could gzip, though, since Erlang can auto decompress gzip'd files. Just some thoughts - willing to discuss.

**Comment by ConnorRigby** [comment]
i considered a bit of preprocessing on it. there's a lot of duplicates in there, weird formatting etc. i don't know how often these get updated, but i assume it's relatively static.

i don't hate the idea of gzipping it either.

**Comment by fhunleth** [comment]
Superseded by #192

**Pull request closed without merging**